### ISSUE TITLE
Do I need to manually position custom overlays? 

### ISSUE THREAD

**Issue by jktravis** [react-bootstrap/react-bootstrap#3095]
Hi. I'm trying to create, what's in effect, something near identical to a `<OverlayTrigger/>` by using the `<Overlay/>` component as recommended in the docs. The only thing different I want is to be able to more finely control when it is shown, potentially by passing in some callbacks.

I can get the custom Overlay to open, but the positioning is all wrong. 

```javascript
<Overlay target={() => this.addButton} show={showAddPopover} placement="top">
  <AddPopover/>
</Overlay>
```

And my target looks like this

```javascript
<Button onClick={this.toggleAddPopover} 
  active={showAddPopover} 
  ref={ref => this.addButton = ref}> Add
</Button>
```

For `container`, I've tried giving it the `body`, `this`, a new div appended to body, the `<Well/>` that the target sits in, and not passing a container prop at all. None of them seem to position it correctly. 

I've seen in the previous issues that the container needs `position:relative`. I've tried that with the container being `this`, and it still doesn't position right. 

Basically, is there a really easy way to just add a couple function callbacks to the OverlayTrigger so that I can show/hide it based on a prop?

**Comment by diegofaria** [comment]
I would like this also.

**Comment by taion** [comment]
make sure your `<AddPopover>` is correctly receiving the `style` props from `<Overlay>`... that's what sets the position

**Comment by lruckman** [comment]
Just to add to this... 

I was experiencing the same issue when I wrapped my Popover in a redux container. The fix was to make sure I passed in the props to Popover in my connected component as @taion mentioned. Specifically the props: placement, positionLeft, positionTop
